server/src/main/java/org/candlepin/async/tasks/HypervisorHeartbeatUpdateJob.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.ArgumentConversionException;
+import org.candlepin.async.JobArguments;
+import org.candlepin.async.JobConfig;
+import org.candlepin.async.JobConfigValidationException;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.common.filter.LoggingFilter;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Owner;
+
+import com.google.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.Objects;
+
+
+
+/**
+ * Asynchronous job for refreshing the last check-in time for a specific reporter ID
+ */
+public class HypervisorHeartbeatUpdateJob implements AsyncJob {
+    private static final Logger log = LoggerFactory.getLogger(HypervisorHeartbeatUpdateJob.class);
+
+    public static final String JOB_KEY = "HYPERVISOR_HEARTBEAT_UPDATE";
+    public static final String JOB_NAME = "hypervisor_heartbeat_update";
+
+    private static final String REPORTER_ID = "reporter_id";
+    private static final String OWNER_KEY = "owner_key";
+
+    private final ConsumerCurator consumerCurator;
+
+    /**
+     * Job configuration object for the export job
+     */
+    public static class HypervisorHeartbeatUpdateJobConfig extends JobConfig {
+        public HypervisorHeartbeatUpdateJobConfig() {
+            this.setJobKey(JOB_KEY)
+                .setJobName(JOB_NAME);
+
+            // TODO: Should this be unique by (owner_key, reporter_id) or owner_key? If it's the
+            // former, we'll need to update the UniqueByArgConstraint to support examining multiple
+            // parameters
```

Comment:
Probably unique by (owner_key, reporter_id), we can add a card about the multi parameter checking.

Reply:
Alright. I assume we'll leave this as the TODO for now until the other card is completed?

Reply:
If you think it's not much work, you can add the multiparameter checking in this PR, otherwise leave the TODO and create a card, up to you.

Reply:
Yeah, I'd rather address the multi-param constraint in another card.